WebdevPeriod3/wwwroot/js/subsystemlist.js

```diff
@@ -0,0 +1,115 @@
+﻿'use strict';
+
+function getButtonRow(buttonEl) { 
+  buttonEl.closest('input'); 
+}
+
+function debounce(callback, wait, immediate = false) {
+  let timeout = null 
+  
+  return function() {
+    const callNow = immediate && !timeout
+    const next = () => callback.apply(this, arguments)
+    
+    clearTimeout(timeout)
+    timeout = setTimeout(next, wait)
+
+    if (callNow) {
+      next()
+    }
+  }
+}
+
+let ctrl = null;
+
+async function search(term) {
+  if (ctrl) ctrl.abort();
+  ctrl = new AbortController();
+
+  const products = await fetch(`../products?searchterm=${term}`, {
```

Comment:
Here you are using a mixture of async and await and promises. Please choose one or the other and do not combine both.